### PR TITLE
Update Supported Ruby Versions

### LINKED ISSUE #97: Update Ruby Support; Drop 2.8, 3.0; Add 3.2, 3.3
Wanted to look at dropping support for Ruby 2.8 (EoL almost a year ago) and 3.0 (expected EoL end of March) and add support for Ruby 3.2, 3.3.

Adding 3.2 does not affect the test suite at all.

There are backtrace related tests affected on Ruby 3.3. A group are easy to fix (`(eval)` was changed to `(eval at filename:lineno)`.) In the other the backtrace is _hiding_ the source of a method block within templates that fail. I need to look into this a little more (I didn't see anything regarding these backtrace changes at a quick glance on the Ruby release notes.)

### PR DESCRIPTION
**Pull request by zacheryph**
> Closes #97

2.7 has been EoL for almost a year. Removing, and adding 3.2. 3.0 Will be EoL in March but decided to leave in place for now.

This also updates all of the development and test dependencies to play nicely with newer versions of Ruby.

All Gemfiles were all updated as versions of Nokogiri and Pry didn't play well with updated Ruby & Bundler packages.

The Rails sha for `rails_master` was _not_ updated. It breaks everything, primarily due to `ActionView::Helpers::TagHelper::TagBuilder::HTML_VOID_ELEMENTS` no longer being a thing at all.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,14 +61,6 @@ Add it to your Gemfile and `bundle install`:
 gem "rbexy"
 ```
 
-_From 1.0 onward, we only support Rails 6. If you're using Rails 5, use the 0.x releases._
-
-In `config/application.rb`:
-
-```ruby
-require "rbexy/rails/engine"
-```
```
Comment (patbenatar) [inline README.md:70]: Why is this removed?

**Comment by patbenatar** [comment]
Thanks for this!

| The Rails sha for rails_master was not updated. It breaks everything, primarily due to ActionView::Helpers::TagHelper::TagBuilder::HTML_VOID_ELEMENTS no longer being a thing at all.

I opened https://github.com/patbenatar/rbexy/issues/101 for this

File: README.md (hunk 1 above)
Comment (zacheryph) [reply README.md:70]: Whoops. This is suppose to be in a different PR I didn't push yet.

**Review by patbenatar — APPROVED** [review]
